tests/unit/test_actions.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s charm actions unit tests."""
+
+import secrets
+
+# Need access to protected functions for testing
+# pylint:disable=protected-access
+import typing
+from unittest.mock import MagicMock
+
+import ops
+import pytest
+
+import charm
+import jenkins
+from charm import JenkinsK8sOperatorCharm
+
+from .types_ import HarnessWithContainer
+
+
+def test__on_get_admin_password_action_container_not_ready(
```

Comment:
Really picky, but how about `test_get_admin_password_action_container_not_ready` to avoid referencing a private method?

Reply:
Done, thanks for the suggestion!

Reply:
I think you forgot to remove the double underscores

Reply:
Woops, done! Thanks!